Src/Base/AMReX_RKIntegrator.H

```diff
@@ -251,7 +251,8 @@ public:
         for (int step_number = 0; step_number < BaseT::max_steps && !stop; ++step_number)
         {
             // Adjust step size to reach output time
-            if (time_out - time_current < dt) {
+            // protect against roundoff
+            if (std::abs(time_out - time_current - dt) <= 1.e5 * std::numeric_limits<amrex::Real>::epsilon() * std::abs(dt)) {
```

Comment:
What if Real is float?

We could also do `amrex::AlmostEqual(time_current+dt, time_out)`. `AlmostEqual` has an optional argument that can be used to fine tune what is considered almost equal.

Reply:
That sounds good, I'll test it.

Reply:
@WeiqunZhang how is this?